GameServerLib/GameObjects/Other/CollisionHandler.cs

```diff
@@ -24,34 +27,41 @@ public CollisionHandler(Game game, IMap map)
         public void AddObject(IGameObject obj)
         {
             _objects.Add(obj);
+            if (obj is IBaseTurret || obj is IInhibitor || obj is INexus)
+            {
+                _quadMap.Insert(obj);
+            }
+            
         }
 
         public void RemoveObject(IGameObject obj)
         {
             _objects.Remove(obj);
+            if (obj is IBaseTurret || obj is IInhibitor || obj is INexus)
+            {
+                _quadMap.Remove(obj);
+            }
         }
 
         public void Update()
         {
+            QuadTree<IGameObject> _quadDynamic = new QuadTree<IGameObject>(20000, 20000, new CollisonObject());
```

Comment:
Any reason as to why we have to rebuild the whole quadtree on every tick? This seems quite inefficient to me, making the benefit of even using a quadtree negligible.

Reply:
QuadTree doesn't change after creation. We discussed it on discord.The benefit is on checking collision. Creation of quad tree seems to me O(Nlog N) or maybe O(N), while the collision check is bounded by 10N and usually less. Instead of O(N^2) we get only O(N log N). Maybe there are better ways to update the quadtree but this lib doesn't support that.